### ISSUE TITLE
Fault Injection connect question

### ISSUE THREAD

**Issue by Wmingyu** [vul337/PrIntFuzz#6]
when i perform fault Injection test，i  failed to connection qemu and reply Boot with module {device} failed!.How can i to solve？

**Comment by ZheyuMa** [comment]
Could you please show me how to reproduce the problem or a screenshot of it to help me locate it?

**Comment by Wmingyu** [comment]
Thank you for reply.My question is step 4 failed.How can i to resolve this question?







At 2024-11-22 21:50:46, "Ma Zheyu" ***@***.***> wrote:

Could you please show me how to reproduce the problem or a screenshot of it to help me locate it?

**Comment by wmy-yyy** [comment]
In the fault injection part, I did not generate a success message, and all were crashes
![1](https://github.com/user-attachments/assets/faea6ecf-121f-470d-aec3-d4deeea3dffb)
![2](https://github.com/user-attachments/assets/ec925400-1509-4fa1-ae90-59d08c5e6cbd)
![3](https://github.com/user-attachments/assets/2a55c1fa-8c45-47a6-b57c-8b3f16e2a7ee)

**Comment by ZheyuMa** [comment]
As acknowledged in the paper, PrIntFuzz’s solution does not guarantee 100% success rate, there will be some cases that fail. You can test it on all drivers and then observe the examples in the success folder.

**Comment by Wmingyu** [comment]
Thank you. We only have 11 successful parts of PCI, most of which are failures, enter a failed emulation device to see that its driver is in use, what is the cause of this situation?
![1](https://github.com/user-attachments/assets/278f4a28-a3b1-472a-ae7f-ad681bd8ae6a)

![2](https://github.com/user-attachments/assets/1840d46f-6597-4c98-b767-3e532a67fac8)

**Comment by ZheyuMa** [comment]
Sorry, I cannot reproduce the problem, everything is fine in my environment. Could you please provide the **completed** log from scratch so that I can locate the problem? BTW, "in use" means that the device emulation was successful, but it was misclassified.

**Comment by Wmingyu** [comment]
We have some problems with the fault injection part. When judging the success or not, "Kernel driver in use" cannot be output, but it can be viewed after entering the virtual device. Could you please explain what caused it?

![image](https://github.com/user-attachments/assets/6a1b9c66-e3a1-4c4d-81ba-ffe335a52c6a)

![image](https://github.com/user-attachments/assets/89d49920-d962-4787-b43a-dd2d4e35651a)

**Comment by ZheyuMa** [comment]
This is weird. It looks like the lspci command reported an error in the first image but not in the second, can you confirm the reason for the error, please?

**Comment by Wmingyu** [comment]
Thank you very much for your reply, but so far I have not found the reason for the error. Let me try again and see if I can solve this problem.

**Comment by Wmingyu** [comment]
I'm sorry to bother you again, but I still have some questions to ask you.
I can't get the system call list for the interrupt section, and I can't find the 'new_fake_{device_name}.txt' and 'new_qemu_{device_name}.txt' files. How can I get the information about the interrupt section?

![image](https://github.com/user-attachments/assets/ec181cbe-2428-40d8-920f-20d40243f2d4)

![image](https://github.com/user-attachments/assets/8a19af05-ead4-4ff3-bb79-3a6ae2cf3218)

![image](https://github.com/user-attachments/assets/c77c4a95-5ba0-443c-b763-86c970b48888)

**Comment by ZheyuMa** [comment]
Thanks for reporting. This version only included the device emulation code, excluding the interrupt system call descriptions, I'll update the code as soon as you needed.

**Comment by Wmingyu** [comment]
Thank you. Now when I go to execute the "fuzz.py -s -f" script, output the message "SYZFAIL: Open print_fuzz failed" in the 'syzkaller.out' file. And should I check the final results in the "syzkaller_workdir/crashes report"?

![image](https://github.com/user-attachments/assets/15108f4b-3e41-4b28-83d2-a172ce604359)


![image](https://github.com/user-attachments/assets/bb72a8c8-23fe-4253-bea4-6897333e26d5)

**Comment by ZheyuMa** [comment]
No, you should check why the "printfuzz" module does not work. You can check whether the module is compiled to the kernel and the permission of the module.

**Comment by Wmingyu** [comment]
I'm sorry to bother you again. I re-did step 7 -"Build the disk image for fuzzing", and I checked for the "print_fuzz.o" file in the "build/linux/linux_fuzz" directory. Meanwhile, I did not find an operation in the python script to add "printfuzz" to the kernel. What should I do to make sure that the "printfuzz" module is included in the kernel? Or how do I check if “printfuzz” is in the kernel?

![image](https://github.com/user-attachments/assets/32215048-4ab5-4250-859f-123682881ace)

**Comment by Wmingyu** [comment]
I'm sorry to bother you again. Could you give us some examples of interruptions, even just one? I would really appreciate it, because it would be very helpful to our work

**Comment by ZheyuMa** [comment]
It looks like the module is compiled in, it could be something else, you need more detailed logs to locate the issue. Please double-check the cause of the error, an error reported by syzkaller is insufficient.

**Comment by ZheyuMa** [comment]
What kind of examples of interruptions do you want? The syzkaller description or something else. BTW, if there are any other questions please start a new issue, it looks clearer, thanks!
